_inc/client/components/jetpack-termination-dialog/dialog.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { translate as __ } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { getSiteBenefits } from 'state/site';
+import { isDevVersion } from 'state/initial-state';
+import analytics from 'lib/analytics';
+import Button from 'components/button';
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+import JetpackTerminationDialogFeatures from 'components/jetpack-termination-dialog/features';
+import QuerySite from 'components/data/query-site';
+import QuerySiteBenefits from 'components/data/query-site-benefits';
+import Spinner from 'components/spinner';
+
+function mapBenefitNameToGridicon( benefitName ) {
+	switch ( benefitName ) {
+		case 'contact-form':
+			return 'align-image-center';
+		case 'contact-form-feedback':
+			return 'mail';
+		case 'image-hosting':
+			return 'image';
+		case 'jetpack-backup':
+			return 'cloud-download';
+		case 'jetpack-stats':
+			return 'stats-alt';
+		case 'protect':
+			return 'lock';
+		case 'publicize':
+			return 'share';
+		case 'sharing':
+			return 'share';
+		case 'subscribers':
+			return 'user';
+		case 'video-hosting':
+			return 'video-camera';
+		default:
+			return 'checkmark';
+	}
+}
+
+function mapBenefitDataToViewData( benefit ) {
+	return {
+		title: benefit.title,
+		description: benefit.description,
+		amount: benefit.value,
+		gridIcon: mapBenefitNameToGridicon( benefit.name ),
+	};
+}
+
+/*
+ * On Jetpack Termination
+ *
+ * This Dialog is designed to be used from multiple locations with different intents, either
+ * disconnecting the Jetpack plugin or uninstalling it. To abstract this we use the word
+ * "Termination" to represent both
+ */
+
+class JetpackTerminationDialog extends Component {
+	static propTypes = {
+		closeDialog: PropTypes.func.isRequired,
+		disconnectJetpack: PropTypes.func.isRequired,
+		location: PropTypes.oneOf( [ 'plugins', 'dashboard' ] ).isRequired,
+		purpose: PropTypes.oneOf( [ 'disconnect', 'disable' ] ).isRequired,
+		siteBenefits: PropTypes.array,
+	};
+
+	componentDidMount() {
+		const { location, purpose } = this.props;
+		analytics.tracks.recordEvent( 'jetpack_termination_dialog_termination_view', {
```

Comment:
Should this be 

```suggestion
		analytics.tracks.recordEvent( 'jetpack_termination_dialog_view', {
```

here and below, or is the double `termination` here on purpose?

Reply:
'tis a typo 🤦‍♂